src/include/btree_cmp_inline.h

```diff
@@ -14,12 +14,188 @@
 #endif
 #endif
 
-#if defined(HAVE_ARM_NEON_INTRIN_H)
-#include <arm_neon.h>
-#endif
-/* 16B alignment */
-#define WT_ALIGNED_16(p) (((uintptr_t)(p)&0x0f) == 0)
 #define WT_VECTOR_SIZE 16 /* chunk size */
+#define WT_COMPARE_SHORT_MAXLEN 16
+
+#ifdef HAVE_X86INTRIN_H
+
+#define WT_VECTOR_MASK 0xFFFF
+/*
+ * __lex_compare_gt_16 --
+ *     Lexicographic comparison routine for data greater than 16 bytes. Returns: < 0 if user_item is
+ *     lexicographically < tree_item, = 0 if user_item is lexicographically = tree_item, > 0 if
+ *     user_item is lexicographically > tree_item. We use the names "user" and "tree" so it's clear
+ *     in the btree code which the application is looking at when we call its comparison function.
+ */
+static WT_INLINE int
+__lex_compare_gt_16(const uint8_t *ustartp, const uint8_t *tstartp, size_t len, int lencmp)
```

Comment:
What you suggested is working and __attribute__((always_inline)) is essential. Otherwise, `__lex_compare_skip_gt_16 ` is not inlined in `__wt_lex_compare`

Reply:
I'm finding it more complicated to have two almost-identical copies of these functions to review and keep in sync. But if you prefer keeping them separate that is OK.

Reply:
Is it still inlining these functions when not deduped? I know compilers generally always inline static functions that are called exactly once, which is why deduping the logic may have caused an out-of-line call.

Reply:
No, `__wt_lex_compare` doesn't inline those functions but `__wt_lex_compare_skip` inlines those functions, which is weird.